.github/workflows/unittest.yml

```diff
@@ -15,21 +15,23 @@ jobs:
         shell: bash -l {0}
     steps:
       - uses: actions/checkout@v4
-      - uses: conda-incubator/setup-miniconda@v3
+      - uses: mamba-org/setup-micromamba@v1
         with:
-          auto-update-conda: true
-          channels: conda-forge,defaults
-          use-mamba: true
           environment-file: environment.yml
-          activate-environment: hyspecplanningtools_dev
+          cache-environment-key: ${{ runner.os }}-env-${{ hashFiles('**/environment.yml') }}
+          cache-downloads-key: ${{ runner.os }}-downloads-${{ hashFiles('**/environment.yml') }}
+          condarc: |
+            channels:
+              - conda-forge
+              - anaconda
       - name: install in editable mode
         run: |
           echo "installing in editable mode"
           python -m pip install -e .
       - name: run unit tests
         run: |
-          echo "running unit tests"
-          python -m pytest --cov=src --cov-report=xml --cov-report=term-missing tests/
+          echo "running unit tests with coverage"
+          xvfb-run --server-args="-screen 0 1920x1080x24" -a python -m pytest --cov=src --cov-report=xml --cov-report=term
```

Comment:
You could use the `pytest-xvfb` package instead if you want [pytest-xvfb](https://pypi.org/project/pytest-xvfb/)